### PR TITLE
Support new_order-style in Certbot

### PR DESCRIPTION
**Pull request by ohemorange**
Part of #5580.

### CODE REVIEW

**Comment by ohemorange** [comment]
Currently adding tests for the new code paths. This should otherwise be ready for review. Also added support for `best_effort`.

File: certbot/client.py [hunk 1]
```diff
@@ -326,10 +318,20 @@ def obtain_certificate(self, domains):
...
             csr = crypto_util.init_save_csr(key, domains, self.config.csr_dir)
 
-        certr, chain = self.obtain_certificate_from_csr(
-            domains, csr, authzr=authzr)
+        orderr = self.acme.new_order(csr.data)
+        authzr = self.auth_handler.handle_authorizations(orderr, self.config.allow_subset_of_names)
+        auth_domains = set(a.body.identifier.value for a in authzr)
+        successful_domains = [d for d in domains if d in auth_domains]
+
+        if successful_domains != domains:
```
Comment (bmw) [inline certbot/client.py]: I don't think you can compare set and list objects like this.

**Comment by bmw** [comment]
LGTM if tests pass.

File: certbot/client.py (hunk 1 above)
Comment (ohemorange) [reply certbot/client.py]: I made the same mistake when testing with ipdb.

**Pull request merged**